client/driver/executor/executor_linux.go

```diff
@@ -41,90 +33,80 @@ var (
 	}
 )
 
-func NewExecutor(ctx *ExecutorContext) Executor {
-	return NewLinuxExecutor(ctx)
-}
-
-func NewLinuxExecutor(ctx *ExecutorContext) Executor {
-	return &LinuxExecutor{ExecutorContext: ctx}
-}
-
-// Linux executor is designed to run on linux kernel 2.8+.
-type LinuxExecutor struct {
-	*ExecutorContext
-	cmd  exec.Cmd
-	user *user.User
-	l    sync.Mutex
-
-	// Isolation configurations.
-	groups   *cgroupConfig.Cgroup
-	taskName string
-	taskDir  string
-	allocDir string
-
-	// Spawn process.
-	spawn *spawn.Spawner
-}
+// configureIsolation configures chroot and creates cgroups
+func (e *UniversalExecutor) configureIsolation() error {
+	if e.ctx.FSIsolation {
+		if err := e.configureChroot(); err != nil {
+			return err
+		}
+	}
 
-func (e *LinuxExecutor) Command() *exec.Cmd {
-	return &e.cmd
+	if e.ctx.ResourceLimits {
+		if err := e.configureCgroups(e.ctx.TaskResources); err != nil {
+			return fmt.Errorf("error creating cgroups: %v", err)
+		}
```

Comment:
Because at this point we haven't started the user process yet so we don't have the pid yet. But since we need to enter only the plugin process to the cgroup I think we can do this here.